services/user/Registry/plugin/src/errors.rs

```diff
@@ -1,37 +1,8 @@
-use crate::bindings::host::common::types::{Error, PluginId};
+use psibase::plugin_error;
 
-#[derive(PartialEq, Eq, Hash)]
-pub enum ErrorType {
-    QueryError,
-    NotFound,
-    QueryDeserializeError,
-}
-
-fn my_plugin_id() -> PluginId {
-    return PluginId {
-        service: "registry".to_string(),
-        plugin: "plugin".to_string(),
-    };
-}
-
-impl ErrorType {
-    pub fn err(self, msg: &str) -> Error {
-        match self {
-            ErrorType::QueryError => Error {
-                code: self as u32,
-                producer: my_plugin_id(),
-                message: format!("Failed to post graphql query: {}", msg),
-            },
-            ErrorType::QueryDeserializeError => Error {
-                code: self as u32,
-                producer: my_plugin_id(),
-                message: format!("Failed to deserialize graphql query response: {}", msg),
-            },
-            ErrorType::NotFound => Error {
-                code: 404,
```

Comment:
So that's where the 404 code was. `plugin_error!()` supports a discriminant specifically for this (`<variant> = <discriminant> => <fmt>`), but then I couldn't find it when I was doing the actual translation. I think Registry must still have been in PR at the time.

Reply:
great. I'll look at the macro.